src/datasets.c

```diff
@@ -1373,7 +1373,7 @@ static int DatasetOpSerialized(Dataset *set, const char *string, DatasetOpFunc D
 
     switch (set->type) {
         case DATASET_TYPE_STRING: {
-            uint32_t decoded_size = SCBase64DecodeBufferSize(strlen(string));
+            uint32_t decoded_size = SCBase64DecodeBufferSize((uint32_t)strlen(string));
```

Comment:
comes from unix socket

Reply:
do we impose some limit?

Reply:
Adding one as there is also the stack allocation below